netbox/resource_netbox_available_ip_address.go

```diff
@@ -125,7 +117,7 @@ func resourceNetboxAvailableIPAddressRead(d *schema.ResourceData, m interface{})
 	}
 
 	if res.GetPayload().AssignedObject != nil {
```

Comment:
Does this check even do anything now?

Reply:
Nope and the other sets are now redundant too.  Removed.  After approving this PR, pls wait for me to squash these commits before you merge

Reply:
Not sure I'm following. I referred to `AssignedObject` being checked for nil but `AssignedObjectId` being used.

Reply:
To expand on this: This resource creates an IP address upstream, right? If you get rid of all  reads, what happens if someone changes the status of your created IP address? Without a proper read function, there will be no config drift detection.

And yes, this is not caught by the current testing standard.

Enlighten me if I'm missing something :smiley:

Reply:
"_config drift_".  Ah, I see the purpose of these read functions now.  Yes I'd just ensured all tests & my manual CRUD checks passed.  I've reinstated and changed that nill check to be for AssignedObjectID, same as is set.